clang/lib/CAS/IncludeTree.cpp

```diff
@@ -743,6 +754,46 @@ llvm::Error IncludeTree::ModuleMap::print(llvm::raw_ostream &OS,
   return forEachModule([&](Module M) { return M.print(OS, Indent); });
 }
 
+llvm::Expected<IncludeTree::APINotes>
+IncludeTree::APINotes::create(ObjectStore &DB,
+                              ArrayRef<ObjectRef> APINoteList) {
+  assert(APINoteList.size() < 2 && "Too many APINotes added");
+  return IncludeTreeBase::create(DB, APINoteList, {});
+}
+
+llvm::Expected<IncludeTree::APINotes>
+IncludeTree::APINotes::get(ObjectStore &DB, ObjectRef Ref) {
+  auto Node = DB.getProxy(Ref);
+  if (!Node)
+    return Node.takeError();
+  if (!isValid(*Node))
+    return llvm::createStringError(llvm::inconvertibleErrorCode(),
+                                   "not an APINodes node kind");
+  return APINotes(std::move(*Node));
+}
+
+llvm::Error IncludeTree::APINotes::print(llvm::raw_ostream &OS,
+                                         unsigned Indent) {
+  return forEachReference([&](ObjectRef Ref) -> llvm::Error {
+    auto Node = getCAS().getProxy(Ref);
+    if (!Node)
+      return Node.takeError();
+    OS.indent(Indent) << Node->getID() << "\n";
+    OS.indent(Indent) << Node->getData() << "\n";
```

Comment:
Is the compiled APINotes human readable? I thought this would be a blob

Reply:
PR has been updated to store uncompiled APINotes (YAML) so it can be printed!